arm-sql/2015-05-01/swagger/sql.json

```diff
@@ -0,0 +1,214 @@
+                    }
+                }
+            },
+            "get": {
+                "tags": [
+                    "SecurityAlertPolicy"
+                ],
+                "operationId": "SecurityAlertPolicy_GetServerPolicy",
+                "description": "Returns an Azure SQL Server security alert policy.",
+                "parameters": [
+                    {
+                      "name": "resourceGroupName",
+                      "in": "path",
+                      "required": true,
+                      "type": "string",
+                      "description": "The name of the Resource Group to which the server belongs."
+                    },
+                    {
+                      "name": "serverName",
+                      "in": "path",
+                      "required": true,
+                      "type": "string",
+                      "description": "The name of the Azure SQL Server."
+                    }, 
+                    {
+                        "$ref": "#/parameters/ApiVersionParameter"
+                    },
+                    {
+                        "$ref": "#/parameters/SubscriptionIdParameter"
+                    }
+                ],
+                "responses": {
+                    "200": {
+                        "description": "The Server Threat Detection Policy",
+                        "schema": {
+                            "$ref": "#/definitions/ServerSecurityAlertPolicy"
+                        }
+                    }
+                }
+            }
+        }
+    },
+  "definitions": {
+    "ServerSecurityAlertPolicyProperties": {
+      "properties": {
+        "state": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The state of the Azure SQL Security alert policy (New, Enabled or Disabled)."
+        },
+        "disabledAlerts": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The disable alerts of the Azure SQL Security alert policy."
+        },
+        "emailAddresses": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The list of email addresses to send the alert."
+        },
+        "emailAccountAdmins": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Value that indicated whether to email the azure account admins."
+        }
+      },
+      "description": "Represents the properties of an Azure SQL Security alert policy."
+    },
+    "ServerSecurityAlertPolicyCreateOrUpdateParameters": {
+      "x-ms-azure-resource": true,
+      "properties": { 
+        "properties": {
+          "$ref": "#/definitions/ServerSecurityAlertPolicyProperties",
+          "description": "Properties of the request."
+        }
+      },
+      "required": [
+          "properties"
+        ],
+      "description": "Create or update server security alert policy parameters."
+    },
+    "ServerSecurityAlertPolicy": {
```

Comment:
Is ServerSecurityAlertPolicy a type of Resource, meaning will it have an ide, name, type, tages (properties of the "Resource" object) ? If so, then ServerSecurityAlertPolicy is actually a child of the Resource class and that should be represented in Swagger with allOf. Take a look at an example of StorageAccount over [here](https://github.com/Azure/azure-rest-api-specs/blob/master/arm-storage/2015-06-15/swagger/storage.json#L632).

One more thing, if the parent class has the "x-ms-azure-resource": true extension then all its child classes will have resource flattening applicable on them.